src/core_atmosphere/dynamics/mpas_atm_time_integration.F

```diff
@@ -1806,12 +1850,24 @@ subroutine advance_scalars(field_name, domain, rk_step, rk_timestep, config_mono
       end if
       if (rk_step < 3 .or. (.not. config_monotonic .and. .not. config_positive_definite)) then
          allocate(horiz_flux_array(num_scalars,nVertLevels,nEdges+1))
-         horiz_flux_array(:,:,nEdges+1) = 0.0_RKIND
+         !$acc parallel default(present)
+         !$acc loop gang vector collapse(2)
+         do iScalar = 1,num_scalars
+            do k = 1, nVertLevels
```

Comment:
Although in practice compilers will probably spot this opportunity for optimization, it could be better for CPU execution, and more in line with how we've written nested loops, to transpose the `iScalar` and `k` loops so that the innermost loop corresponds to the left-most array dimension.

Also, though a minor detail, I'd suggest consistently using a space after the comma in the loop bounds (e.g., `1, num_scalars` rather than `1,num_scalars`).

Reply:
Good point. Fixed it in the new commit.